subworkflows/nf-core/vcf_phase_shapeit5/main.nf

```diff
@@ -0,0 +1,88 @@
+include { BCFTOOLS_INDEX as VCF_INDEX2      } from '../../../modules/nf-core/bcftools/index/main.nf'
+
+workflow VCF_PHASE_SHAPEIT5 {
+
+    take:
+    ch_vcf        // channel (mandatory): [ val(meta), path(vcf), path(csi), path(pedigree), val(region) ]
+    ch_ref        // channel (optional) : [ val(meta), path(ref), path(csi) ]
+    ch_scaffold   // channel (optional) : [ val(meta), path(scaffold), path(csi) ]
+    ch_map        // channel (optional) : [ val(meta), path(map)]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    // It is needed to generate a file containing the region to phase in a Chr \tab Start \tab End format
+    // The meta map needing to be conserved the following steps a required
+
+    // Keep the meta map and the region in two separated channel but keed id field to link them back
+    ch_region = ch_vcf
+        .multiMap { m, vcf, csi, pedigree, region ->
+            metadata: [ m.id, m]
+            region  : [ m.id, region]
+        }
+
+    // Create the File in bed format and use the meta id for the file name
+    ch_merged_region = ch_region.region
+        .collectFile { mid, region -> [ "${mid}.bed", region.replace(":","\t").replace("-","\t") ] }
+        .map { file -> [ file.baseName, file ] }
+
+    // Link back the meta map with the file
+    ch_region_file = ch_region.metadata
+        .join(ch_merged_region, failOnMismatch:true, failOnDuplicate:true)
+        .map { mid, meta, region_file -> [meta, region_file] }
+
+    BEDTOOLS_MAKEWINDOWS(ch_region_file)
+    ch_versions = ch_versions.mix(BEDTOOLS_MAKEWINDOWS.out.versions.first())
+
+    ch_chunk_output = BEDTOOLS_MAKEWINDOWS.out.bed
+        .splitCsv(header: ['Chr', 'Start', 'End'], sep: "\t", skip: 0)
+        .map { meta, it -> [meta, it["Chr"]+":"+it["Start"]+"-"+it["End"]] }
+        .view()
+
+    // Count the number of chunks
+    chunks_number = ch_chunk_output
+        .groupTuple()
+        .map { meta, chunks -> [meta, chunks.size()]}
+
+    ch_phase_input = ch_vcf
+        .map { m, vcf, index, pedigree, region ->
+            [m, vcf, index, pedigree] }
+        .combine(ch_chunk_output, by:0)
+
+    SHAPEIT5_PHASECOMMON ( ch_phase_input,
+                            ch_ref,
+                            ch_scaffold,
+                            ch_map )
+    ch_versions = ch_versions.mix(SHAPEIT5_PHASECOMMON.out.versions.first())
+
+    VCF_INDEX1(SHAPEIT5_PHASECOMMON.out.phased_variant)
+    ch_versions = ch_versions.mix(VCF_INDEX1.out.versions.first())
+
+    ch_ligate_input = SHAPEIT5_PHASECOMMON.output.phased_variant
+        .combine(chunks_number, by:0)
+        .map{meta, vcf, chunks_num -> [groupKey(meta, chunks_num), vcf]}
+        .groupTuple()
+        .combine(VCF_INDEX1.out.csi
+            .combine(chunks_number, by:0)
+            .map{meta, csi, chunks_num -> [groupKey(meta, chunks_num), csi]}
+            .groupTuple(),
+            by:0)
```

Comment:
```suggestion
        .join(VCF_INDEX1.out.csi, failOnMismatch:true, failOnDuplicate:true)
        .combine(chunks_number, by:0)
        .map{meta, vcf, csi, chunks_num -> [groupKey(meta, chunks_num), vcf, csi]}
        .groupTuple()
```
This will do the same, but less verbose and a bit more flexible

Reply:
You'll probably also need to create a temp ID field which you will change back afterwards to make sure the chunk files have different names (this is a good practice too for traceability)

Reply:
Shouldn't it be made at the nextflow.config level ?
If I understand, I should modify the meta.id in the phasing process (to add the region) and set it back to the origin after the ligation ?

Reply:
Yes that would be ideal! We try to not use the config as much as possible in subworkflows, because those values can be easily overwritten which would break the whole workflow :)

Reply:
Allright! Have a nice meal 😋